src/config/backend.ts

```diff
@@ -64,6 +73,12 @@ export abstract class DeviceConfigBackend {
 
 	// Return the env var name for this config option
 	public abstract createConfigVarName(configName: string): string;
+
+	// Allow a chosen config backend to be initialised
+	public initialise(opts: BackendOptions): Promise<DeviceConfigBackend> {
+		this.options = { ...this.options, ...opts };
+		return Promise.resolve(this);
```

Comment:
It'd be nicer to just make the function `async` and then this can be `return this` and it'll be automatically wrapped, also any exceptions will also get automatically wrapped